core/sail/fts/lucene-api/src/main/java/org/eclipse/rdf4j/sail/lucene/LuceneSail.java

```diff
@@ -282,14 +282,20 @@ public SearchIndex getLuceneIndex() {
 	public NotifyingSailConnection getConnection()
 		throws SailException
 	{
-		return new LuceneSailConnection(super.getConnection(), luceneIndex, this);
+		if (!closed.get()) {
+			return new LuceneSailConnection(super.getConnection(), luceneIndex, this);
+		}
+		else {
+			throw new SailException("Connection cannot be taken from shutdown sail");
+		}
 	}
 
 	@Override
 	public void shutDown()
 		throws SailException
 	{
-		if(closed.compareAndSet(false, true)) {
+		logger.debug("shutdown");
```

Comment:
We need to be a little careful with what kind of debug logging we put in place - it's a big framework and it can quickly get messy and overwhelming.  Perhaps rephrase to "LuceneSail shutDown".